spec.bs

```diff
@@ -1717,3 +1764,147 @@ specification is printed below:
   /fenced-frame/cspee.https.html
   /fenced-frame/embedder-csp-not-propagate.https.html
 </wpt>
+
+<h3 id=permissions-policy-changes>Permissions Policies</h3>
+
+Permissions are granted to {{Document}} inside of <{fencedframe}>s through the
+{{FencedFrameConfig}} object and its associated internal [=fencedframeconfig/config=], which defines
+the permissions [=fenced frame config/required permissions to load|required=] for a <{fencedframe}>
+to navigate successfully. Specifically, a {{Document}} inside of a <{fencedframe}> can only load if
+it has opted into all of the [=fenced frame config/required permissions to load=].
+
+<h4 id=fenced-container-policies>Fenced container policies</h4>
+
+TODO: define <dfn>fenced container policy</dfn> here, which is current referenced by
+<{fencedframe/allow}>.
+
+<h4 id=permissions-policy-patches>Algorithm patches</h4>
+
+<div algorithm=create-permissions-policy>
+  Modify the definition of [=Create a Permissions Policy for a navigable=] to read:
+
+  Given null or an element (|container|), an [=origin=] (<var ignore>origin</var>), and an optional
+  [=list=] of [=policy-controlled features=] (|required features|), this algorithm returns a new
+  [=Document/permissions policy=].
+
+  Rewrite step 1 of the algorithm to read:
+
+  1. [=Assert=]: if not null, |container| is either a [=navigable container=] or a [=fenced
+     navigable container=].
+
+  Add a new step after step 1 that reads:
+
+  2. [=Assert=]: if |container| is not a [=fenced navigable container=], |required features| is not
+     given.
+
+  Rewrite step 4 to read:
+
+  4. If |container| is a [=fenced navigable container=], then for each |feature| supported:
+
+       1. If |feature| [=list/exists=] in |required features|, set |inherited policy|[|feature|] to
```

Comment:
I think this is broken as-is right now. Your later modifications to https://html.spec.whatwg.org/multipage/document-sequences.html#creating-a-new-browsing-context pass in nothing as the `|required policies|` variable, but it seems that when `|container|` is a fenced navigable container, we expect `|required features|`. We should fix this and in the process decide whether:
 - The initial `about:blank` Document in a fenced frame should have _all_ policies manually disabled; or...
 - Or inherit them from "beyond the fence" as iframes would normally do

The first option seems like the safest, and is kinda not web-observable except for with extensions I guess. But I'd love to know what our implementation does.

Reply:
I agree with doing option 1. That will match the behavior in our implementation once the required permissions CL is landed. If an empty permissions list is passed into PermissionsPolicy::CreateForFencedFrame, it will disable every feature.